src/limitedmap.h

```diff
@@ -26,68 +26,76 @@ class limitedmap
     typedef typename std::multimap<V, iterator>::iterator rmap_iterator;
     size_type nMaxSize;
 
+private:
+    rmap_iterator find_rmap_key_value_pair(const_iterator map_it) {
+        auto [it, it_last] = rmap.equal_range(map_it->second);
+        for (; it != it_last; ++it) {
+            if (it->second == map_it) {
+                return it;
+            }
+        }
+        return rmap.end();
+    }
+
 public:
-    limitedmap(size_type nMaxSizeIn = 0) { nMaxSize = nMaxSizeIn; }
+    LimitedMap(size_type nMaxSizeIn = 0) { nMaxSize = nMaxSizeIn; }
```

Comment:
I honestly don't understand the rational for letting this object be created with a max_size==0
It would prevent insertion of any element.
IMHO the default shold be removed and drop an assertion for values <1

Reply:
Not sure what the rationale was, but I agree we don't have any use case for that. I applied your suggestion.